### PR TITLE
Allow Generator object for seed in QuantumVolume circuit

### LINKED ISSUE #4866: Allow passing in a Generator object for seed in QuantumVolume circuit
### What is the expected enhancement?

Allow `QuantumVolume` to be called with a generator object rather than only using a numerical seed. This is important for sharing RNG objects in applications using these circuits (such as in Ignis), and also internally in the QV circuit we can pass the RNG object to the `random_unitary` function which already accepts a generator for the seed kwarg.

Example: to generate a list of quantum volume circuits from a single seed value:
```python
seed = 101
rng = np.random.default_rng(seed)
circs = [QuantumVolume(10, 10, seed=rng) for _ in range(20)]
```

### PR DESCRIPTION
**Pull request by chriseclectic**
### Summary

Allows passing in a Numpy random `Generator` object as seed value for `QuantumVolume` circuit.

Closes #4866

### CODE REVIEW

**Comment by nonhermitian** [comment]
It would be nice if the circuit name had the seed and the offset from this seed in the name. Then each circuit could be reconstructed from the name alone.

**Comment by chriseclectic** [comment]
@nonhermitian As far as i know this info can't be extract from the PCG64 Generator since it doesn't actually use an integer seed internally but using the int to generate a SeedSequence to generate a BitGenerator.

**Comment by nonhermitian** [comment]
I am basically advocating for something like I did here:

https://github.com/Qiskit/qiskit-terra/blob/81216e55b339ef0cdab50d43d116eeb261dff540/qiskit/circuit/library/quantum_volume.py#L95

where the seed used to start the generator is attached to the circuit, plus the number of times the generator was called.  Flor example if the seed was `2231`, and I just seeded (offset=0) it then the name would be like `QuantumVolume_10x10_2231+0`.

With this convention I can reproduce the circuit, I would just re-seed with the same seed, and then call offset number of times before generating the circuit.

File: qiskit/circuit/library/quantum_volume.py [hunk 1]
```diff
@@ -62,45 +62,44 @@ class QuantumVolume(QuantumCircuit):
...
             perm = rng.permutation(perm_0)
             if not classical_permutation:
                 layer_perm = Permutation(num_qubits, perm)
                 inner.compose(layer_perm, inplace=True)
             for w in range(width):
-                seed_u = unitary_seeds[d][w]
-                su4 = random_unitary(4, seed=seed_u).to_instruction()
-                su4.label = 'su4_' + str(seed_u)
+                su4 = random_unitary(4, seed=rng).to_instruction()
+                su4.label = 'su4'
```
Comment (ajavadia) [inline qiskit/circuit/library/quantum_volume.py:102]: don't remove the label, it was intentionally added here so it is displayed like this
```
                             ┌──────────┐
q_0: ────────────────────────┤1         ├
     ┌──────────┐┌──────────┐│  su4_210 │
q_1: ┤1         ├┤1         ├┤0         ├
     │  su4_912 ││  su4_534 │└──────────┘
q_2: ┤0         ├┤0         ├────────────
     └──────────┘└──────────┘
```

your change just makes it display like
```
     ┌──────┐┌──────┐
q_0: ┤0     ├┤1     ├────────
     │      ││      │┌──────┐
q_1: ┤  su4 ├┤  su4 ├┤1     ├
     │      ││      ││  su4 │
q_2: ┤1     ├┤0     ├┤0     ├
     └──────┘└──────┘└──────┘
```

Comment (chriseclectic) [reply qiskit/circuit/library/quantum_volume.py:102]: I can change it back, but to do so I would have to remove passing the generator object to the `random_unitary` functions. This kind of defeats the point of using a shared generator object in the first place.

File: releasenotes/notes/quantum_volume_rng-2e6f46e3821aebeb.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+upgrade:
```
Comment (mtreinish) [inline releasenotes/notes/quantum_volume_rng-2e6f46e3821aebeb.yaml:2]:
```suggestion
features:
```

**Pull request merged**